Hunting Queries/W3CIISLog/WebShellActivity.yaml

```diff
@@ -0,0 +1,59 @@
+id: e0c947c3-fe83-46ff-bbda-a43224a785fd
+name: Web Shell Activity
+description: |
+  'Web shells are scripts that, when uploaded to a web server, can be used to provide a backdoor to a compromised network.
+  Attackers can use this entry point to leave malicious implants, such as obtaining unauthorized access, escalating privilege, and further compromising the environment. 
+  
+  This query hunts for web shells by analysing the distribution of commonly-used web shell scripts againts regular scripts for those public client IPs which have not observed any W3CIIS activity in a fixed lookback period.
+  The results obtained summarise the public client IPs, user agents, and the distribution of the above scripts between the start and end time.'
+requiredDataConnectors:
+  - connectorId: AzureMonitor(IIS)
+    dataTypes:
+      - W3CIISLog
+tactics:
+  - Persistence
+  - InitialAccess
+relevantTechniques:
+  - T1100
+query: | 
+
+  let end_time = now();
+  let start_time = end_time - timespan(1d);
+  let lookback_time_from_start_time = timespan(3d);
+  let script_extensions = dynamic([".asp", ".aspx", ".armx", ".asax", ".ashz", ".asmx", ".axd", ".cshtml", ".php", ".phps", ".php3", ".php4", ".php5", ".php7", ".jsp", ".jspx", ".cfm", ".cfml", ".phtml"]);
+  let ignore_uristems = dynamic(["/ews/exchange.asmx"]);
+  let lookback_period = (
+  W3CIISLog 
+  | where TimeGenerated between ((start_time - lookback_time_from_start_time).. start_time)
+  | where not(ipv4_is_private(cIP)) and cIP != "127.0.0.1"
+  | summarize count() by cIP, csUserAgent
+  | project cIP, csUserAgent
+  );
+  let potential_webshell_activity = (W3CIISLog
+  | where TimeGenerated between (start_time .. end_time)
+  | extend csUriStem = tolower(csUriStem)
+  | where csUriStem matches regex \\.[a-zA-Z][a-zA-Z0-9]+$
```

Comment:
Your regex needs to be enclosed in quotes here

Reply:
Resolved